subworkflows/local/busco_diamond_blastp.nf

```diff
@@ -0,0 +1,83 @@
+//
+//
+// Run BUSCO for a genome from GOAT and runs diamond_blastp
+//
+//
+
+nextflow.enable.dsl = 2
+
+include { GOAT_TAXONSEARCH    } from '../../modules/local/goat_taxon_search'
+include { BUSCO               } from '../../modules/nf-core/modules/busco/main'
+include { EXTRACT_BUSCO_GENES } from '../../modules/local/extract_busco_genes'
+//include { DIAMOND_BLASTP      } from '../../modules/nf-core/modules/diamond/blastp/main'
+
+workflow BUSCO_DIAMOND {
+    take:
+
+    //  Tuple [meta, fasta]:
+    fasta
+
+    main:
+
+    ch_versions = Channel.empty()
+
+
+    //
+    // Fetch BUSCO lineages for taxon (or taxa)
+    //
+
+    GOAT_TAXONSEARCH (
+    fasta.map { fa -> [fa[0], "${params.taxon}", "${params.taxa_file}" ? file("${params.taxa_file}") : []] }
+    )
+    ch_versions = ch_versions.mix(GOAT_TAXONSEARCH.out.versions)
+
+    //
+    // Run BUSCO search
+    //
+
+    // Make a new channel in which each output file is replaced by its content, with meta propagated
+    ch_lineages = GOAT_TAXONSEARCH.out.busco_lineages.flatMap { it[1].readLines().collect { line -> [it[0], line] } }
+    // Cross-product of both channels, using meta as the key
+    ch_busco_inputs = fasta.combine(ch_lineages, by: 0)
+
+    BUSCO (
+    ch_busco_inputs.map { [it[0], it[1]] },
+    ch_busco_inputs.map { it[2] },
+    "${params.busco_lineages_path}",  // Please pass this option. We don't want to download the lineage data every time.
+    [] // No config
+    )
+    ch_versions = ch_versions.mix(BUSCO.out.versions)
+
+    //
+    // Extract BUSCO genes
+    //
+
+    dir = BUSCO.out.busco_dir.map { fileid, path -> path }
+    tables = Channel.fromPath( ["$dir/**/run_archaea_odb10/full_table.tsv", "$dir/**/run_bacteria_odb10/full_table.tsv", "$dir/**/run_eukaryota_odb10/full_table.tsv"] )
```

Comment:
Does this mean that the rest of this subworkflow doesn't care about the other lineages ? Are the other lineages used in another workflow ? If so, they need to be listed in an output (`emit:`) channel of this subworkflow.

Reply:
The `full_table.tsv` for the first lineage in the lineages file (`carnivora_odb10 `  for _Meles meles_) is an input for `diamond_blastx subworkflow` and should be an output channel as well.